application/modules/custom_fields/models/Mdl_custom_fields.php

```diff
@@ -262,7 +264,7 @@ public function get_values_for_fields($custom_field_model, $model_id)
                         $field->$key_serialized .= $custom_value === end($custom_values) ? '' : ', ';
                     }
                 }
-            } else {
+            } elseif ($field->custom_field_type == 'SINGLE-CHOICE') {
```

Comment:
I didn't test this, but is there a reason why you didn't keep an else statement?
Or is that causing the bug to re-occur, even if you added the elseif?

Reply:
I did not write the function itself but as far as I understand `get_values_for_fields()` only handles values for choice fields (multiple or single). The else statement handles all custom field types which leads to wrong values, because there are no values to get for types like text or dates.

Reply:
Oke, sounds fair :)